server/pulp/plugins/util/metadata_writer.py

```diff
@@ -148,17 +150,28 @@ def _open_metadata_file_handle(self):
         else:
             self.metadata_file_handle = open(self.metadata_file_path, 'w')
 
-    def _write_file_header(self):
+    def _write_xml_header(self):
         """
-        Write any headers for the metadata file
+        Write the initial <?xml?> header tag into the file handle.
         """
-        pass
+        assert self.metadata_file_handle is not None
```

Comment:
Nope.

I should've mentioned it in the commit message perhaps, but I tried really hard to do a straight refactor and avoid mixing in a bunch of behavior changes that would make it difficult to separate the refactor itself from any improvements, as well as to avoid introducing new and fun regressions.